From ec757960b81458ac3f68aac5a43f769734d01803 Mon Sep 17 00:00:00 2001
From: Zenghui Yu <yuzenghui@huawei.com>
Date: Thu, 11 Jan 2024 20:43:14 +0800
Subject: [PATCH 539/960] downstream: KVM: arm64: vtimer: Expose HW-based
 vtimer interrupt in debugfs

Expose HW-based vtimer interrupt in debugfs

Signed-off-by: Zenghui Yu <yuzenghui@huawei.com>
Signed-off-by: Kunkun Jiang <jiangkunkun@huawei.com>
Signed-off-by: Dongxu Sun <sundongxu3@huawei.com>
Signed-off-by: Slim6882 <15605196882@163.com>
Signed-off-by: chenyi <chenyi211@h-partners.com>
Signed-off-by: Liu Shixin <liushixin2@huawei.com>
---
 arch/arm64/kvm/vgic/vgic-debug.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm64/kvm/vgic/vgic-debug.c b/arch/arm64/kvm/vgic/vgic-debug.c
index 07aa0437125a..20fed597531b 100644
--- a/arch/arm64/kvm/vgic/vgic-debug.c
+++ b/arch/arm64/kvm/vgic/vgic-debug.c
@@ -193,7 +193,7 @@ static void print_irq_state(struct seq_file *s, struct vgic_irq *irq,
 		print_header(s, irq, vcpu);
 
 	pending = irq->pending_latch;
-	if (irq->hw && vgic_irq_is_sgi(irq->intid)) {
+	if (vgic_direct_sgi_or_ppi(irq)) {
 		int err;
 
 		err = irq_get_irqchip_state(irq->host_irq,
-- 
2.43.0

